### PR TITLE
Adding supporting for Azure AD access token

### PR DESCRIPTION
**Pull request by yitam**
This change is [Reviewable](https://reviewable.io/reviews/microsoft/msphpsql/837)

### CODE REVIEW

File: source/shared/core_conn.cpp [hunk 1]
```diff
@@ -759,35 +765,53 @@ void build_connection_string_and_set_conn_attr( _Inout_ sqlsrv_conn* conn, _Inou
...
-            if( pwd != NULL ) {
-                escaped = core_is_conn_opt_value_escaped( pwd, strnlen_s( pwd ));
-                CHECK_CUSTOM_ERROR( !escaped, conn, SQLSRV_ERROR_UID_PWD_BRACES_NOT_ESCAPED ) {
+        // Add the server name
+        common_conn_str_append_func( ODBCConnOptions::SERVER, server, strnlen_s( server ), connection_string TSRMLS_CC );
+       
+        // if uid is not present then we use trusted connection -- but not when access token is used, because they are incompatible
+        if (!access_token_used) {
+            if (uid == NULL || strnlen_s(uid) == 0) {
+                connection_string += CONNECTION_OPTION_NO_CREDENTIALS;  //  "Trusted_Connection={Yes};"
```
Comment (david-puglielli) [inline source/shared/core_conn.cpp]: Where is CONNECTION_OPTION_NO_CREDENTIALS defined?

File: source/shared/core_conn.cpp [hunk 2]
```diff
@@ -1172,3 +1196,54 @@ size_t core_str_zval_is_true( _Inout_ zval* value_z )
...
+    //  typedef struct AccessToken
+    //  {
+    //      unsigned int dataSize;
+    //      char data[];
+    //  } ACCESSTOKEN;
+    //
+    // NOTE: The ODBC Driver version 13.1 only supports this authentication on Windows.
+    //
+    // A valid access token byte string must be expanded so that each byte is followed by a 0 padding byte, 
+    // similar to a UCS-2 string containing only ASCII characters
```
Comment (david-puglielli) [inline source/shared/core_conn.cpp]: It sounds like the foregoing comments are from a Microsoft Docs document - if so please provide a link to it.

File: source/pdo_sqlsrv/pdo_dbh.cpp [hunk 3]
```diff
@@ -38,6 +38,7 @@ namespace PDOConnOptionNames {
 
 const char Server[] = "Server";
 const char APP[] = "APP";
+const char AccessToken[] = "AccessToken";
```
Comment (david-puglielli) [inline source/pdo_sqlsrv/pdo_dbh.cpp]: Is this an ODBC keyword as well?

File: source/shared/core_conn.cpp (hunk 1 above)
Comment (yitam) [reply source/shared/core_conn.cpp]: At the beginning of the same file. It was there all along but wasn't used.

File: source/shared/core_conn.cpp (hunk 2 above)
Comment (yitam) [reply source/shared/core_conn.cpp]: ok will do

File: source/pdo_sqlsrv/pdo_dbh.cpp (hunk 3 above)
Comment (yitam) [reply source/pdo_sqlsrv/pdo_dbh.cpp]: No, but similar to TraceFile and TraceOn, this one is also a pre-connection attribute.

**Pull request merged**